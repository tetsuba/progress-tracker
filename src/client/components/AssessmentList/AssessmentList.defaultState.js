import type { AssessmentType } from './AssessmentList.types'

export const ASSESSMENT_LIST: Array<AssessmentType> = [
  {
    assessmentId: 'alphabet-upperCase',
    description: 'Upper case letter assessment',
    path: `/student/{id}/assessment/alphabet-upperCase`,
    title: 'Alphabet Uppercase',
  },
  {
    assessmentId: 'alphabet-lowerCase',
    description: 'Lower case letter assessment',
    path: `/student/{id}/assessment/alphabet-lowerCase`,
    title: 'Alphabet Lowercase',
  },
  {
    assessmentId: 'alphabet-names',
    description: 'Name the letters assessment',
    path: `/student/{id}/assessment/alphabet-names`,
    title: 'Alphabet Names',
  },
  {
    assessmentId: 'alphabet-sounds',
    description: 'Make the letter sounds assessment',
    path: `/student/{id}/assessment/alphabet-sounds`,
    title: 'Alphabet Sounds',
  },
]

export const ASSESSMENT_NOT_FOUND = {
  assessmentId: 'assessment-not-found',
  description: 'Assessment not found',
  path: '',
  title: 'Assessment not found',
}
