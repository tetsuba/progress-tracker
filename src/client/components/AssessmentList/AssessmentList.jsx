import React from 'react'
import { CardDeck } from 'react-bootstrap'

// COMPONENTS
import AssessmentCard from '../AssessmentCard/AssessmentCard'

// DEFAULT STATE
import {
  ASSESSMENT_LIST,
  ASSESSMENT_NOT_FOUND,
} from './AssessmentList.defaultState'

// TYPES
import type { AssessmentType } from './AssessmentList.types'

export function findAssessmentDetails(
  assessmentName: string,
  id: string
): AssessmentType {
  const assessment = ASSESSMENT_LIST.find(
    ({ assessmentId }) => assessmentId === assessmentName
  )
  return assessment
    ? {
        ...assessment,
        path: assessment.path.replace('{id}', id),
      }
    : ASSESSMENT_NOT_FOUND
}

function getAssessmentsNotActive(
  assessments: Array<string>
): Array<AssessmentType> {
  return ASSESSMENT_LIST.filter(
    ({ assessmentId }) => !assessments.includes(assessmentId)
  )
}

type Props = {
  assessments: Array<string>,
}
export default function AssessmentList(props: Props) {
  const { assessments } = props
  const assessmentsNotActive = getAssessmentsNotActive(assessments)

  return (
    <CardDeck className="mt-5">
      {assessmentsNotActive.map((assessmentProps, index) => (
        <AssessmentCard key={`card-${index}`} assessment={assessmentProps} />
      ))}
    </CardDeck>
  )
}
