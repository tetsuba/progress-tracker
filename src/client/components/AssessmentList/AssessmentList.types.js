export type AssessmentType = {
  assessmentId: string,
  title: string,
  description: string,
  path: string,
}
