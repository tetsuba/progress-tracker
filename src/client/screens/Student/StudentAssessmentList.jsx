import React from 'react'
import { useHistory } from 'react-router-dom'
import { ListGroup } from 'react-bootstrap'
import { findAssessmentDetails } from '../../components/AssessmentList/AssessmentList'

type Props = {
  assessments: [string],
  id: string,
}

export default function StudentAssessmentList(props: Props) {
  const { assessments, id } = props
  const history = useHistory()

  return (
    <ListGroup>
      {assessments.map((assessment) => {
        const { title, path } = findAssessmentDetails(assessment, id)
        return (
          <ListGroup.Item
            style={{ cursor: 'pointer' }}
            key={title}
            onClick={() => history.push(path)}
          >
            {title}
          </ListGroup.Item>
        )
      })}
    </ListGroup>
  )
}
