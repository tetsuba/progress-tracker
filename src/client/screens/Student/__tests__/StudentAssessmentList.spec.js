import { testRendererFull } from '../../../../test/testHelper'
import StudentAssessmentList from '../StudentAssessmentList'
import { ASSESSMENT_LIST } from '../../../components/AssessmentList/AssessmentList.defaultState'

jest.mock('react-router-dom', () => ({
  useHistory: jest.fn(),
}))

describe('<StudentAssessmentList>', () => {
  describe('@Render', () => {
    it('Should render all assessments', () => {
      const props = {
        assessments: ASSESSMENT_LIST.map(({ assessmentId }) => assessmentId),
        id: 'studentId01',
      }
      const wrapper = testRendererFull(StudentAssessmentList, props)

      expect(wrapper.find('ListGroupItem')).toHaveLength(
        props.assessments.length
      )
      expect(wrapper).toMatchSnapshot()
    })
  })
})
