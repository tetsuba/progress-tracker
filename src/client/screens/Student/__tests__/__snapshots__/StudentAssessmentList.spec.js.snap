// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StudentAssessmentList> @Render Should render all assessments 1`] = `
<StudentAssessmentList
  assessments={
    Array [
      "alphabet-upperCase",
      "alphabet-lowerCase",
      "alphabet-names",
      "alphabet-sounds",
    ]
  }
  id="studentId01"
>
  <ListGroup
    horizontal={null}
    variant={null}
  >
    <ForwardRef
      as="div"
      className="list-group"
      onSelect={[Function]}
    >
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <ListGroupItem
          active={false}
          disabled={false}
          key="Alphabet Uppercase"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
          variant={null}
        >
          <ForwardRef
            as="div"
            className="list-group-item"
            disabled={false}
            eventKey={null}
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Alphabet Uppercase
            </div>
          </ForwardRef>
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          key="Alphabet Lowercase"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
          variant={null}
        >
          <ForwardRef
            as="div"
            className="list-group-item"
            disabled={false}
            eventKey={null}
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Alphabet Lowercase
            </div>
          </ForwardRef>
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          key="Alphabet Names"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
          variant={null}
        >
          <ForwardRef
            as="div"
            className="list-group-item"
            disabled={false}
            eventKey={null}
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Alphabet Names
            </div>
          </ForwardRef>
        </ListGroupItem>
        <ListGroupItem
          active={false}
          disabled={false}
          key="Alphabet Sounds"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
          variant={null}
        >
          <ForwardRef
            as="div"
            className="list-group-item"
            disabled={false}
            eventKey={null}
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Alphabet Sounds
            </div>
          </ForwardRef>
        </ListGroupItem>
      </div>
    </ForwardRef>
  </ListGroup>
</StudentAssessmentList>
`;
